companyweb_base/models/res_partner.py

```diff
@@ -0,0 +1,532 @@
+    cweb_creditLimit_info_visible = fields.Boolean(
+        "Companyweb Credit limit Info Visible", readonly=True
+    )
+
+    cweb_startDate = fields.Date("Companyweb Start Date", readonly=True)
+    cweb_startDate_enable = fields.Boolean(
+        "Companyweb Start Date Enabled", readonly=True
+    )
+    cweb_endDate = fields.Date("Companyweb End Date", readonly=True)
+    cweb_endDate_enable = fields.Boolean("Companyweb End date Enabled", readonly=True)
+    cweb_score = fields.Char("Companyweb Score", readonly=True)
+    cweb_score_enable = fields.Boolean("Companyweb Score Enabled", readonly=True)
+
+    cweb_image_tag = fields.Html(
+        "Companyweb Barometer Image tag", compute="_compute_cweb_image", readonly=True
+    )
+    cweb_image = fields.Char("Companyweb Barometer Image", readonly=True)
+    cweb_image_visible = fields.Boolean(
+        "Companyweb Barometer Image Visible", readonly=True
+    )
+
+    cweb_warnings = fields.Html("Companyweb Warnings", readonly=True)
+    cweb_warnings_enable = fields.Boolean("Companyweb Warnings Enabled", readonly=True)
+    cweb_url = fields.Char("CompanyWeb Detailed Report", readonly=True)
+    cweb_url_enable = fields.Boolean(
+        "CompanyWeb Detailed Report Enabled", readonly=True
+    )
+    cweb_url_report = fields.Char("CompanyWeb Url Report", readonly=True)
+    cweb_url_report_enable = fields.Boolean(
+        "CompanyWeb Url Report Enabled", readonly=True
+    )
+    cweb_vat_liable = fields.Boolean("CompanyWeb Subject to VAT", readonly=True)
+    cweb_vat_liable_enable = fields.Boolean(
+        "CompanyWeb Subject to VAT Enabled", readonly=True
+    )
+    cweb_balance_data_enable = fields.Boolean(
+        "CompanyWeb Balance Data Enabled", readonly=True
+    )
+    cweb_balance_year = fields.Char("CompanyWeb Balance Year", readonly=True)
+    cweb_balance_year_visible = fields.Boolean(
+        "CompanyWeb Balance Year Visible", readonly=True
+    )
+    cweb_closed_date = fields.Date("CompanyWeb Closed Date", readonly=True)
+    cweb_closed_date_visible = fields.Boolean(
+        "CompanyWeb Closed Date Visible", readonly=True
+    )
+    cweb_equityCapital = fields.Float("CompanyWeb Equity Capital", readonly=True)
+    cweb_equityCapital_visible = fields.Boolean(
+        "CompanyWeb Equity Capital Visible", readonly=True
+    )
+    cweb_average_fte = fields.Float(
+        "CompanyWeb Average number of staff in FTE", readonly=True
+    )
+    cweb_average_fte_visible = fields.Boolean(
+        "CompanyWeb Average number of staff in FTE Visible", readonly=True
+    )
+    cweb_addedValue = fields.Float("CompanyWeb Gross Margin (+/-)", readonly=True)
+    cweb_addedValue_visible = fields.Boolean(
+        "CompanyWeb Gross Margin (+/-) Visible", readonly=True
+    )
+    cweb_turnover = fields.Float("CompanyWeb Turnover", readonly=True)
+    cweb_turnover_visible = fields.Boolean("CompanyWeb Turnover Visible", readonly=True)
```

Comment:
Here we need this boolean because Odoo does not distingish NULL and 0 for numerical fields.
But, still nitpicking ;), I'd name that `cweb_turnover_unset` or something which is closer to the actual semantics of the field. The fact that it is visible or not is already a derivative use of the field.